Jenkinsfile

```diff
@@ -51,20 +51,61 @@ pipeline {
                 sh "make init-qemu"
             }
         }
-        stage('Lint and Test') {
+        stage('Vendor') {
+            steps {
+                sh """
+                    mkdir -p .cargo || /bin/true
+                    make vendor
+                """
+            }
+        }
+        stage('Lint and Unit Test') {
+            parallel {
+                stage('Lint'){
+                    steps {
+                        withCredentials([[
+                                           $class: 'AmazonWebServicesCredentialsBinding',
+                                           credentialsId: 'aws',
+                                           accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                                           secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+                                         ]]){
+                            sh """
+                              make lint
+                            """
+                        }
+                    }
+                }
+                stage('Unit Tests'){
+                    steps {
+                        withCredentials([[
+                                           $class: 'AmazonWebServicesCredentialsBinding',
+                                           credentialsId: 'aws',
+                                           accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                                           secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+                                         ]]){
+                            sh """
+                              make -j2 test
+                            """
+                        }
+                    }
+                }
+            }
+        }
+        stage('Test') {
             environment {
                 CREDS_FILE = credentials('pipeline-e2e-creds')
                 LOGDNA_HOST = "logs.use.stage.logdna.net"
             }
             parallel {
-                stage('Lint, Unit and Integration Tests'){
+                stage('Integration Tests'){
                     steps {
                         script {
                             def creds = readJSON file: CREDS_FILE
                             // Assumes the pipeline-e2e-creds format remains the same. Chase
                             // refer to the e2e tests's README's authorization docs for the
                             // current structure
                             LOGDNA_INGESTION_KEY = creds["packet-stage"]["account"]["ingestionkey"]
+                            TEST_THREADS = sh (script: 'echo "$(nproc)/4"| bc', returnStdout: true).trim()
```

Comment:
$ echo "$(nproc)/10"| bc
0

for
$ nproc
8

TEST_THREADS=0 works?

Reply:
Nice catch, I added an extra bit of logic to make sure it's never 0. I couldn't just add 1 as it meant machines with 4 cores like ec2-fleet would try to use 2 threads, which I found was prone to breakage, but I added a slightly ugly ternary